pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -101,79 +93,15 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 		podChanged = true
 	}
 
-	// Sync source code to the component
-	// If syncing for the first time, sync the entire source directory
-	// If syncing to an already running component, sync the deltas
-	// If syncing from an odo watch process, skip this step, as we already have the list of changed and deleted files.
-	if !podChanged && !parameters.ForceBuild && len(parameters.WatchFiles) == 0 && len(parameters.WatchDeletedFiles) == 0 {
```

Comment:
Are we potentially breaking `odo watch` here? `odo watch` passes in `WatchFiles` and `WatchDeletedFiles` into the adapter's Push function, but I don't see those files getting passed in to the new sync adapter.